pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -0,0 +1,133 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	"istio.io/istio/pkg/log"
+)
+
+/***************************  Mesh EDS Implementation **********************************/
+// Implements EDS interface methods of v2.MeshDiscovery
+
+// StreamEndpoints implements xdsapi.EndpointDiscoveryServiceServer.StreamEndpoints().
+func (s *DiscoveryServer) StreamEndpoints(stream xdsapi.EndpointDiscoveryService_StreamEndpointsServer) error {
+	ticker := time.NewTicker(responseTickDuration)
+	peerInfo, ok := peer.FromContext(stream.Context())
+	peerAddr := "Unknown peer address"
+	if ok {
+		peerAddr = peerInfo.Addr.String()
+	}
+	defer ticker.Stop()
+	var discReq *xdsapi.DiscoveryRequest
+	var receiveError error
+	var clusters []string
+	initialRequest := true
+	reqChannel := make(chan *xdsapi.DiscoveryRequest, 1)
+	go func() {
+		defer close(reqChannel)
+		for {
+			req, err := stream.Recv()
+			if err != nil {
+				if status.Code(err) == codes.Canceled || err == io.EOF {
+					return
+				}
+				receiveError = err
+				log.Errorf("request loop for EDS for client %q terminated with errors %v", peerAddr, err)
+				return
+			}
+			reqChannel <- req
+		}
+	}()
+	for {
+		// Block until either a request is received or the ticker ticks
+		select {
+		case discReq, ok = <-reqChannel:
+			if !ok {
+				return receiveError
+			}
+			if !initialRequest {
+				// Given that Pilot holds an eventually consistent data model, Pilot ignores any acknowledgements
+				// from Envoy, whether they indicate ack success or ack failure of Pilot's previous responses.
+				// Only the first request is actually horored and processed. Pilot drains all other requests from this stream.
+				if log.DebugEnabled() {
+					log.Debugf("EDS ACK from Envoy for client %q has version %q and Nonce %q for request for clusters %v",
+						discReq.GetVersionInfo(), discReq.GetResponseNonce(), clusters)
+				}
+				continue
+			}
+		case <-ticker.C:
+			if !initialRequest {
+				// Ignore ticker events until the very first request is processed.
+				continue
+			}
+		}
+		if initialRequest {
+			initialRequest = false
+			clusters = discReq.GetResourceNames()
+			if len(clusters) == 0 {
+				return fmt.Errorf("no clusters specified in EDS request from %q", peerAddr)
+			}
+			if log.DebugEnabled() {
```

Comment:
Can you add an extra env variable (since global debugEnabled is too broad, and likely nobody knows how to fine-control the new logging framework), and based on that log at info level more detail info - including this and all updates we send ?

Reply:
Uh ... I don't agree. Why should this PR a subject of that change? If you feel the strategy around logs is flawed, that ought to be taken up with @geeknoid and folks who introduced logs and any fixes ought to be done Istio wide.

Also having a patchwork of environment variables for these sorts of things already adds to our confused and muddled test frameworks. This makes for a horrible UX and I'd like to hear really strong objections before we throw out the work that's been done for logging.

Reply:
Ok, then figure out how the logging is supposed to selectively enable this component, and add a comment :-)